.github/workflows/update_virus_defs_production.yml

```diff
@@ -0,0 +1,155 @@
+          CURRENT_DAILY_CVD=`jq -r '.tags[] | select(.Key=="daily.cvd") | .Value' /tmp/ecr_tags.json`
+          CURRENT_MAIN_CVD=`jq -r '.tags[] | select(.Key=="main.cvd") | .Value' /tmp/ecr_tags.json`
+
+          echo "CURRENT_BYTECODE_CVD=$CURRENT_BYTECODE_CVD" >> $GITHUB_ENV
+          echo "CURRENT_DAILY_CLD=$CURRENT_DAILY_CLD" >> $GITHUB_ENV
+          echo "CURRENT_DAILY_CVD=$CURRENT_DAILY_CVD" >> $GITHUB_ENV
+          echo "CURRENT_MAIN_CVD=$CURRENT_MAIN_CVD" >> $GITHUB_ENV
+          
+      - name: Get S3 virus def tags
+        run: |
+          S3_BYTECODE_CVD=`aws s3api get-object-tagging --region ca-central-1 --bucket $S3_BUCKET --key clamav_defs/bytecode.cvd | jq -r '.TagSet[] | select(.Key=="md5") | .Value'`
+          S3_DAILY_CLD=`aws s3api get-object-tagging --region ca-central-1 --bucket $S3_BUCKET --key clamav_defs/daily.cld | jq -r '.TagSet[] | select(.Key=="md5") | .Value'`
+          S3_DAILY_CVD=`aws s3api get-object-tagging --region ca-central-1 --bucket $S3_BUCKET --key clamav_defs/daily.cvd | jq -r '.TagSet[] | select(.Key=="md5") | .Value'`
+          S3_MAIN_CVD=`aws s3api get-object-tagging --region ca-central-1 --bucket $S3_BUCKET --key clamav_defs/main.cvd | jq -r '.TagSet[] | select(.Key=="md5") | .Value'`
+
+          echo "S3_BYTECODE_CVD=$S3_BYTECODE_CVD" >> $GITHUB_ENV
+          echo "S3_DAILY_CLD=$S3_DAILY_CLD" >> $GITHUB_ENV
+          echo "S3_DAILY_CVD=$S3_DAILY_CVD" >> $GITHUB_ENV
+          echo "S3_MAIN_CVD=$S3_MAIN_CVD" >> $GITHUB_ENV
+
+      - name: Check if image needs to be updated
+        id: update_avdefs
+        shell: bash
+        run: |
+          if [[ $CURRENT_BYTECODE_CVD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $CURRENT_DAILY_CLD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $CURRENT_DAILY_CVD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $CURRENT_MAIN_CVD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $S3_BYTECODE_CVD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $S3_DAILY_CLD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $S3_DAILY_CVD =~ ^[a-f0-9]{32}$ ]] && \
+            [[ $S3_MAIN_CVD =~ ^[a-f0-9]{32}$ ]]
+          then
+              if [ "$CURRENT_BYTECODE_CVD" != "$S3_BYTECODE_CVD" ]; then
+                echo ::set-output name=update_avdefs::true
+              elif [ "$CURRENT_DAILY_CLD" != "$S3_DAILY_CLD" ]; then
+                echo ::set-output name=update_avdefs::true
+              elif [ "$CURRENT_DAILY_CVD" != "$S3_DAILY_CVD" ]; then
+                echo ::set-output name=update_avdefs::true
+              elif [ "$CURRENT_MAIN_CVD" != "$S3_MAIN_CVD" ]; then
+                echo ::set-output name=update_avdefs::true
+              else
+                echo ::set-output name=update_avdefs::false
+              fi
+          else
+            echo ::set-output name=update_avdefs::false
+          fi
+
+  build-push-and-deploy:
+    runs-on: ubuntu-latest
+    needs: download-avdefs
+    if: needs.download-avdefs.outputs.update_avdefs
```

Comment:
Does this properly coerce the output string `true` and `false`?  You might have to be explicit like so:
```yaml
if: needs.download-avdefs.outputs.update_avdefs == "true"
```

Reply:
not sure but i'll switch since that's easier to read